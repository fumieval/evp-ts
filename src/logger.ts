import pc from 'picocolors';

export interface ILogger {
    info(message: string) : void;
    error(message: string): void;
}

export class ConsoleLogger implements ILogger {
    public info(
        message: string,
    ): void {
        console.info(`${process.stdout.isTTY ? pc.blue('[EVP]') : '[EVP]'} ${message}`);
    }
    public error(
        message: string,
    ): void {
        console.error(`${process.stderr.isTTY ? pc.red('[EVP]') : '[EVP]'} ${message}`);
    }
}

export function logMissingVariable(logger: ILogger, key: string): void {
    logger.error(`${key}=undefined ERROR: missing environment variable`);
}
