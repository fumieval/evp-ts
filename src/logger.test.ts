import { describe, it, expect, vi, beforeEach } from 'vitest';
import { ConsoleLogger, ILogger } from './logger';

describe('Logger', () => {
  let logger: ILogger;

  beforeEach(() => {
    logger = new ConsoleLogger();
  });

  describe('log', () => {
    it('should log messages correctly', () => {
      const consoleSpy = vi.spyOn(console, 'log').mockImplementation(() => {});
      
      logger.success('KEY', 'value', false);
      
      expect(consoleSpy).toMatchSnapshot();
      consoleSpy.mockRestore();
    });
  });

  describe('error', () => {
    it('should log errors correctly', () => {
      const consoleSpy = vi.spyOn(console, 'error').mockImplementation(() => {});
      
      logger.error('KEY', 'value', new Error('test error'));
      
      expect(consoleSpy).toMatchSnapshot();
      consoleSpy.mockRestore();
    });
  });
});