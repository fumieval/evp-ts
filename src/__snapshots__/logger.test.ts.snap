// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Logger > error > should log errors correctly 1`] = `
[MockFunction error] {
  "calls": [
    [
      "[EVP] KEY=value ERROR: test error",
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`Logger > log > should log messages correctly 1`] = `[MockFunction log]`;
